test/replication-luatest/linearizable_read_test.lua

```diff
@@ -0,0 +1,273 @@
+local t = require('luatest')
+local cluster = require('test.luatest_helpers.cluster')
+local server = require('test.luatest_helpers.server')
+local g = t.group('linearizable-read')
+
+local proxy = require('test.luatest_helpers.proxy.proxy')
+
+local function build_replication(num_instances)
+    local t = {}
+    for i = 1, num_instances do
+        table.insert(t, server.build_instance_uri('server_' .. i .. '_proxy'))
+    end
+    return t
+end
+
+local num_servers = 3
+
+g.before_all(function()
+    g.cluster = cluster:new({})
+    g.servers = {}
+    local box_cfg = {
+        replication = build_replication(num_servers),
+        replication_timeout = 0.1,
+        txn_isolation = 'read-confirmed',
+        memtx_use_mvcc_engine = true,
+    }
+    g.servers[1] = g.cluster:build_and_add_server({
+        alias = 'server_1',
+        box_cfg = box_cfg,
+    })
+    -- Servers 2 and 3 are interconnected without a proxy.
+    for i = 2, num_servers do
+        box_cfg.replication[i] = server.build_instance_uri('server_' .. i)
+    end
+    for i = 2, num_servers do
+        g.servers[i] = g.cluster:build_and_add_server({
+            alias = 'server_' .. i,
+            box_cfg = box_cfg,
+        })
+    end
+    g.proxies = {}
+    for i = 1, num_servers do
+        g.proxies[i] = proxy:new({
+            client_socket_path = server.build_instance_uri('server_' .. i .. '_proxy'),
+            server_socket_path = server.build_instance_uri('server_' .. i),
+        })
+        g.proxies[i]:start({force = true})
+    end
+    g.cluster:start()
+    g.cluster:wait_fullmesh()
+    g.servers[1]:exec(function()
+        box.schema.space.create('sync', {is_sync=true})
+        box.space.sync:create_index('pk')
+    end)
+    g.servers[2]:wait_vclock_of(g.servers[1])
+    g.servers[3]:wait_vclock_of(g.servers[1])
+end)
+
+g.after_all(function()
+    g.cluster:drop()
+end)
+
+g.test_wait_others = function(g)
+    g.proxies[1]:pause()
+    local fid = g.servers[1]:exec(function()
+        local t = require('luatest')
+        local f = require('fiber').new(function()
+            return pcall(box.begin, {linearizable = true})
+        end)
+        f:set_joinable(true)
+        t.assert_equals(f:status(), 'suspended', 'begin() waits for replicas')
+        return f:id()
+    end)
```

Comment:
I would suggest to add here a little fiber sleep for, say, 0.01 ms and then check that the fiber is not finished yet. That it is actually waiting. Otherwise this test will pass even if `box.begin()` succeeds immediately.

Reply:
Good idea, done:
```diff
diff --git a/test/replication-luatest/linearizable_read_test.lua b/test/replication-luatest/linearizable_read_test.lua
index bc8a3ae58..494d107ba 100644
--- a/test/replication-luatest/linearizable_read_test.lua
+++ b/test/replication-luatest/linearizable_read_test.lua
@@ -64,10 +64,12 @@ g.test_wait_others = function(g)
     g.proxies[1]:pause()
     local fid = g.servers[1]:exec(function()
         local t = require('luatest')
-        local f = require('fiber').new(function()
+        local fiber = require('fiber')
+        local f = fiber.new(function()
             return pcall(box.begin, {linearizable = true})
         end)
         f:set_joinable(true)
+        fiber.sleep(0.01)
         t.assert_equals(f:status(), 'suspended', 'begin() waits for replicas')
         return f:id()
     end)
```